### PR TITLE
kernel: crypto: add kmod-crypto-chacha20poly1305 for strongswan

### LINKED ISSUE #18192: strongswan: chacha20poly1305 not working
Maintainer: @pprindeville, @Thermi
Environment: x86_64 and ramips, openwrt master branch

Description:
Strongswan on Openwrt was not able to set up a SA with chacha20poly1305 or chacha20poly1305-x25519, and the log said:
`daemon.info ipsec: 10[KNL] received netlink error: Function not implemented`
while strongswan on debian works well and wireguard on openwrt which uses chacha20poly1305-x25519 as well with no problem.

### PR DESCRIPTION
**Pull request by xwang1498**
Needed by strongSwan IPsec VPN for `strongswan-mod-chapoly`. Not to be confused with kmod-crypto-LIB-chacha20poly1305, which is an 8-byte nonce version used by wireguard.

When this change is accepted, `strongswan-mod-chapoly` package can be updated to depend on this kmod.

This will fix [#18192](https://github.com/openwrt/packages/issues/18192).

reporter: @ru67tm
strongswan package maintainers: @pprindeville @Thermi

Please **also backport this to 22.03** because strongswan-mod-chapoly was first introduced there.

Compile & run tested: mvebu, WRT3200ACM, `22.03-SNAPSHOT, r19381-b42511c007`
Compile & run tested: mvebu, WRT3200ACM, `SNAPSHOT, r19704-efff48529b`

### CODE REVIEW

**Review by pprindeville — APPROVED** [review]
LGTM

**Comment by Thermi** [comment]
If it builds it LGTM

**Review by Thermi — APPROVED** [review]

**Comment by pprindeville** [comment]
See also https://github.com/openwrt/packages/pull/18654.

**Comment by xwang1498** [comment]
Recent committers to crypto.mk, if you could kindly approve & merge these changes if they look good: @dangowrt @pepe2k  @hauke @rmilecki @blocktrron **Thank you!**

**Comment by stintel** [comment]
Merged, thank.

**Comment by xwang1498** [comment]
@stintel Could you please also apply this change to the 22.03 branch? strongswan-mod-chapoly will remain broken there without this fix.

I've compile & run tested this fix on 22.03.

**Comment by pprindeville** [comment]
> Merged, thank.

Thanks!

**Comment by stintel** [comment]
> @stintel Could you please also apply this change to the 22.03 branch? strongswan will remain broken there without this fix.
> 
> I've compile & run tested this fix on 22.03.

Done

**Pull request closed without merging**
